pkg/steps/template.go

```diff
@@ -519,84 +531,101 @@ func waitForPodCompletion(ctx context.Context, podClient PodClient, namespace, n
 	return pod, nil
 }
 
-func waitForPodCompletionOrTimeout(ctx context.Context, podClient PodClient, namespace, name string, completed map[string]time.Time, notifier ContainerNotifier, skipLogs bool) (*coreapi.Pod, error) {
-	// Warning: this is extremely fragile, inherited legacy code.  Please be
-	// careful and test thoroughly when making changes, as they very frequently
-	// lead to systemic production failures.  Some guidance:
-	// - There is a complex interaction between this code and the container
-	//   notifier.  Updates to the state of the pod are received via the watch
-	//   and communicated to the notifier.  Even in case of interruption (i.e.
-	//   cancellation of `ctx`) and/or failure, events should continue to be
```

Comment:
the existing code doesn't match this comment.  When the context is cancelled, the for loop hits  branch that insta-returns.